### PR TITLE
Test that 'prevent immediate propagation' is set

### LINKED ISSUE #7516: Payment Request algorithms tests wrong wrt stop immediate propagation
The spec says: 

> Set event's stop propagation flag and stop immediate propagation flag.

But the following tests assume that flag is not set: 
https://w3c-test.org/payment-request/algorithms-manual.https.html

### PR DESCRIPTION
**Pull request by marcoscaceres**
* closes #7516

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@rsolomakhin, @domenic, this hopefully addresses the issue we were discussing in https://bugs.chromium.org/p/chromium/issues/detail?id=763784#c1

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-09-28 06:56:20
Finished: 2017-09-28 07:03:05


<h2>Failing Jobs</h2>
<ul>

<li>chrome:unstable</li>

<li>MicrosoftEdge:14.14393</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/18937)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/280742165)

File: payment-request/algorithms-manual.https.html [hunk 1]
```diff
@@ -70,52 +70,33 @@
...
       };
     });
-    const p2 = new Promise(resolve => {
-      request.addEventListener(eventName, event => {
-        events.push(event);
-        resolve("listener-1");
-      });
+    // This listener should never fire because the
+    // the event handler caused "prevent immediate propagation" to be set.
+    request.addEventListener(eventName, async () => {
```
Comment (domenic) [inline payment-request/algorithms-manual.https.html]: Would it make sense to use `t.unreached_func("Second event listener should never fire")`?

Comment (marcoscaceres) [reply payment-request/algorithms-manual.https.html]: Oh, nice! I didn't know about `t.unreached_func`!

**Review by domenic — APPROVED** [review]

**Pull request merged**